homeassistant/components/climate/warmup4ie.py

```diff
@@ -0,0 +1,232 @@
+        """Return the polling state."""
+        return True
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._device.get_current_temmperature()
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._device.get_target_temmperature()
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._device.get_target_temperature_low()
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._device.get_target_temperature_high()
+
+    @property
+    def is_away_mode_on(self):
+        """Return if away mode is on."""
+        return self._away
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        return self._on
+
+    @property
+    def operation_list(self):
+        """Return the operation modes list."""
+        return self._operation_list
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. manual, auto, frost."""
+        return self._current_operation_mode
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        if kwargs.get(ATTR_TEMPERATURE) is not None:
+            self._device.set_new_temperature(kwargs.get(ATTR_TEMPERATURE))
+
+    def turn_away_mode_on(self):
+        """Turn away mode on."""
+        self._away = True
+        self._device.set_location_to_frost()
+
+    def turn_away_mode_off(self):
+        """Turn away mode off."""
+        self._away = False
+        self._device.set_temperature_to_manual()
+
+    def set_operation_mode(self, operation_mode):
+        """
+        Set new target operation mode.
+
+        Switch device on if was previously off
+        """
+        if not self.is_on:
+            self._on = True
+        if operation_mode == STATE_AUTO:
+            self._device.set_temperature_to_auto()
+            self._current_operation_mode = operation_mode
+            return
+        if operation_mode == STATE_MANUAL:
+            self._device.set_temperature_to_manual()
+            self._current_operation_mode = operation_mode
+            return
+
+    def turn_on(self):
+        """Turn on."""
+        self._on = True
+        self._device.set_temperature_to_manual()
+
+    def turn_off(self):
+        """Turn off."""
+        self._on = False
+        self._device.set_location_to_off()
+
+    def update(self):
+        """Fetch new state data for this device.
+
+        This is the only method that should fetch new data for Home Assistant.
+        """
+        self._device.update_room()
```

Comment:
I assume that there will be a traceback if the device is offline.

Reply:
Please clarify. Since I am no native speaker I have difficulties to understand what you mean. If updating of the device fails, then simply the internal data (that is temperature, operation_mode) are not updated but will still be available.

Reply:
Okay, i have changed the method. Now an error is logged if fetching new data has failed.